File: chat-service/src/main/java/com/muco/chatservice/domain/persistence/entity/Chatroom.java
package com.muco.chatservice.domain.persistence.entity;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Table(name = "chatroom")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Chatroom {

    @Id
    private Long id;

    private String name;

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: chat-service/src/main/java/com/muco/chatservice/domain/persistence/entity/Chatroom.kt
package com.muco.chatservice.domain.persistence.entity

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.annotation.LastModifiedDate
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table(name = "chatroom")
data class Chatroom(
        @Id
        val id: Long,
        val name: String,
        @CreatedDate
        val createdAt: LocalDateTime,
        @LastModifiedDate
        val updatedAt: LocalDateTime
) {}
